pages/docs/manual/latest/global-special-values.mdx

```diff
@@ -0,0 +1,148 @@
+---
+title: "Global Special Values"
+description: "Global special values for debugging purposes"
+canonical: "/docs/manual/latest/global-special-values"
+---
+
+# Global special values
+
+The ReScript standard library exposes a set of values and functions with useful information for debugging, like file or module names, the current line number, etc.
+
+They are easily recognizable as they all start and end with two dashes, e.g. `__MODULE__`.
+
+### Location
+
+`__LOC__` returns the location in the file at which you put this very expression.
+
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+__LOC__->Js.log
+```
+
+```js
+console.log("File \"testmodule.res\", line 1, characters 0-7");
+```
+
+</CodeTab>
+
+
+### File
+
+`__FILE__` returns the name of the file currently being parsed by the compiler.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+__FILE__->Js.log
+```
+
+```js
+console.log("testmodule.res");
+```
+
+</CodeTab>
+
+### Line
+
+`__LINE__` returns the line number in the file at which you put this very expression.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+// Line 1
+// Line 2
+__LINE__->Js.log
+```
+
+```js
+console.log(3);
+```
+
+</CodeTab>
+
+### Module
+
+`__MODULE__` returns the module name of the file. Note that it does not return the name of a nested module, though.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+__MODULE__->Js.log
+```
+
+```js
+console.log("Testmodule");
+```
+
+</CodeTab>
+
+### Position
+
+`__POS__` returns a tuple consisting of `(file_name, line_number, column_number_start, column_number_end)` corresponding to the location in the file at which you put this very expression.
+
+It basically provides the same information as `__LOC__`, but separated into multiple values. 
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+/* padding comment */ __POS__->Js.log
+```
+
+```js
+console.log(["testmodule.res", 1, 22, 29]);
+```
+
+</CodeTab>
+
+### Location of
+
+`__LOC_OF__(expression)` returns a tuple consisting of `(location, expression)` corresponding to the location of the expression you pass as a parameter here.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+__LOC_OF__(() => None)->Js.log;
```

Comment:
According to my previous comment, destructuring could help newcomers. Not inlining the function demonstrates a common usage.
```suggestion
let f = () => None
let (loc, f') = __LOC_OF__(f)

loc->Js.log
f'->Js.log
```
The js tab would need to be adapted accordingly.